datajunction-server/datajunction_server/api/nodes.py

```diff
@@ -144,26 +147,43 @@ async def validate_node(
     )
 
 
-@router.post("/nodes/{name}/validate/", response_model=NodeValidation)
+@router.post("/nodes/{name}/validate/", response_model=NodeStatusDetails)
 async def revalidate(
     name: str,
     session: AsyncSession = Depends(get_session),
     current_user: Optional[User] = Depends(get_current_user),
-) -> JSONResponse:
+) -> NodeStatusDetails:
     """
     Revalidate a single existing node and update its status appropriately
     """
-    status = await revalidate_node(
+    node_validator = await revalidate_node(
         name=name,
         session=session,
         current_user=current_user,
     )
-    return JSONResponse(
-        status_code=HTTPStatus.OK,
-        content={
-            "message": f"Node `{name}` has been set to {status}",
-            "status": status,
-        },
+
+    return NodeStatusDetails(
```

Comment:
Why the change in the return type?

Reply:
The new return type allows us to display the detailed status in the UI -- instead of just returning a `VALID` or `INVALID`, this actually has a list of errors if the status is invalid.